### PR TITLE
Improved machine authentication article

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Improves machine authentication article

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/9036

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: src/main/pages/user-management/authentication.md [hunk 1]
```diff
@@ -1,70 +1,31 @@
...
+### Auth Filters on master
+
  Che master can handle both types of signed requests - using Keycloak token or Machine token.
- Authentication tokens can be send either in a `Authorization` header or `token` query param.
+ Authentication tokens can be send in a `Authorization` header. Also, in limited cases when it's not possible to use `Authorization` header, token can be send in `token` query parameter. An example of it is initialization iframe where authentication is needed, IDE shows java doc in this way.
 
  In case of Keycloak tokens, authentication filter chain contains of two filters:
 
-- **org.eclipse.che.multiuser.keycloak.server.KeycloakAuthenticationFilter** - validates signature of the Keycloak JWT token;
+- **org.eclipse.che.multiuser.keycloak.server.KeycloakAuthenticationFilter** - validate's signature of the Keycloak JWT token;
```
Comment (slemeur) [inline src/main/pages/user-management/authentication.md]: 'validates' was fine ;)

**Review by slemeur — APPROVED** [review]

**Comment by ghost** [comment]
@sleshchenko merge when it's time

File: src/main/pages/user-management/authentication.md [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
 {% include links.html %}
 
-## Auth Filters on master
+## Authentication on Che Master
+
+### Auth Filters on master
+
  Che master can handle both types of signed requests - using Keycloak token or Machine token.
- Authentication tokens can be send either in a `Authorization` header or `token` query param.
+ Authentication tokens can be send in a `Authorization` header. Also, in limited cases when it's not possible to use `Authorization` header, token can be send in `token` query parameter. An example of it is initialization iframe where authentication is needed, IDE shows java doc in this way.
```
Comment (skabashnyuk) [inline src/main/pages/user-management/authentication.md]: An example of such exceptional case can be: OAuth authentification initialization, IDE shows **javadoc** ...

File: src/main/pages/user-management/authentication.md [hunk 3]
```diff
@@ -22,49 +25,7 @@ folder: user-management
...
-Contains information about the algorithm which the token was signed<br>
-`CHE_MACHINE_AUTH_SIGNATURE__ALGORITHM`<br>
-Contains public key value encoded in Base64<br>
-`CHE_MACHINE_AUTH_SIGNATURE__PUBLIC__KEY`
-
-
-## Obtaining Keycloak endpoints
+### Obtaining Keycloak endpoints
 
 There is separate REST service to obtain main Keycloak endpoints for the current installation. It helps to clients to find out URLs for basic operations like login/logout, profile reading, realm name etc. This service is not secured with any authentication and accessible for any client. Service URL is: `http://<wsmaster_host>:<port>/api/keycloak/settings`
```
Comment (skabashnyuk) [inline src/main/pages/user-management/authentication.md]: ```It helps to clients to find``` - rephrase please

File: src/main/pages/user-management/authentication.md [hunk 4]
```diff
@@ -106,6 +67,63 @@ curl
...
 
+## Authentication on Che Agents
+
+Machines may contain services that must be protected with authentication, e.g. agents like workspace agent and terminal. For this purpose, machine authentication mechanism should be used. Machine tokens were introduced to avoid passing the Keycloak tokens to the machine side (which can be potentially insecure). Another reason is that Keycloak tokens may have relatively small lifetime and require periodical renewal/refresh which is hard to manage and keep in sync with same user session tokens on clients etc.
+
+As agents cannot be queried using Keycloak token, there is only Machine Token option. Machine token can be also passed in header or query parameter.
+
+### Machine JWT Token
+
+Machine token is JWT that contains the following information in its claim:
```
Comment (skabashnyuk) [inline src/main/pages/user-management/authentication.md]: Link to JWT spec or something that explains what it is.

**Pull request merged**